app/models/formula.rb

```diff
@@ -15,4 +15,17 @@
 #
 class Formula < ApplicationRecord
   belongs_to :calculator
+  validate :fields_are_included_in_formulas
+
+  private
+
+  def fields_are_included_in_formulas
+    field_names = calculator.fields.map(&:var_name)
+    formula_variables = expression.scan(/\b[a-zA-Z_]\w*\b/).uniq
+    unused_fields = formula_variables.reject { |var| field_names.include?(var) }
+
+    return unless unused_fields.any?
```

Comment:
ну навіщо так ускладнювати
```suggestion
    return if unused_fields.blank?
```